src/components/simulator/input/d3Chart/disposable/D3ChartDisposable.tsx

```diff
@@ -0,0 +1,143 @@
+import { D3ChartCandlesticksProps } from 'components/simulator/input/d3Chart/D3ChartCandlesticks';
+import { DragablePriceRange } from 'components/simulator/input/d3Chart/recurring/DragablePriceRange';
+import { handleStateChange } from 'components/simulator/input/d3Chart/recurring/utils';
+import { isZero } from 'components/strategies/common/utils';
+import { ScaleLinear } from 'd3';
+import { useCallback, useEffect, useRef } from 'react';
+import { prettifyNumber } from 'utils/helpers';
+
+type Props = Pick<
+  D3ChartCandlesticksProps,
+  'prices' | 'onPriceUpdates' | 'dms' | 'onDragEnd'
+> & {
+  yScale: ScaleLinear<number, number>;
+  isLimit: { buy: boolean; sell: boolean };
+  readonly?: boolean;
+};
+
+export const D3ChartDisposable = ({
+  prices,
+  onPriceUpdates,
+  dms,
+  yScale,
+  isLimit,
+  readonly,
+  onDragEnd,
+}: Props) => {
+  const onMinMaxChange = useCallback(
+    (type: 'buy' | 'sell', min: number, max: number) => {
+      const minInverted = yScale.invert(min).toString();
+      const maxInverted = yScale.invert(max).toString();
+
+      const buy = {
+        min: type === 'buy' ? minInverted : prices.buy.min,
+        max: type === 'buy' ? maxInverted : prices.buy.max,
+      };
+      const sell = {
+        min: type === 'sell' ? minInverted : prices.sell.min,
+        max: type === 'sell' ? maxInverted : prices.sell.max,
+      };
+      onPriceUpdates({ buy, sell });
+    },
+    [
+      onPriceUpdates,
+      prices.buy.max,
+      prices.buy.min,
+      prices.sell.max,
+      prices.sell.min,
+      yScale,
+    ]
+  );
+
+  const hasDragEnded = useRef(false);
+
+  const type =
+    isZero(prices.buy.min) && isZero(prices.buy.max) ? 'sell' : 'buy';
+
+  const onMinMaxChangeEnd = useCallback(
+    (type: 'buy' | 'sell', y1?: number, y2?: number) => {
+      const minInverted = y1
+        ? yScale.invert(y1).toString()
+        : type === 'buy'
+        ? prices.buy.min
+        : prices.sell.min;
+
+      const maxInverted = y2
+        ? yScale.invert(y2).toString()
+        : type === 'buy'
+        ? prices.buy.max
+        : prices.sell.max;
+
+      const buy = {
+        min: type === 'buy' ? minInverted : prices.buy.min,
+        max: type === 'buy' ? maxInverted : prices.buy.max,
+      };
+      const sell = {
+        min: type === 'sell' ? minInverted : prices.sell.min,
+        max: type === 'sell' ? maxInverted : prices.sell.max,
+      };
+      onDragEnd?.({ buy, sell });
+      hasDragEnded.current = true;
+    },
+    [
+      onDragEnd,
+      prices.buy.max,
+      prices.buy.min,
+      prices.sell.max,
+      prices.sell.min,
+      yScale,
+    ]
+  );
+
+  const labels = {
+    buy: {
+      min: prettifyNumber(prices.buy.min, { decimals: 4 }),
```

Comment:
prettifyNumber(value, { decimals: 4}) happens a lot throughout the app. Consider creating a utils function to keep the label number formatting consistent

Reply:
I don't think this is a good idea to create a wrapper of `prettifyNumber` just for one param in the option. I find it already difficult to work with all the number function, I think it'll just make it harder.